e2e/tests/upgrades/upgrade_test.go

```diff
@@ -405,3 +551,29 @@ func getICAVersion(chainAVersion, chainBVersion string) string {
 	// explicitly set the version string because the host chain might not yet support incentivized channels.
 	return icatypes.NewDefaultMetadataString(ibctesting.FirstConnectionID, ibctesting.FirstConnectionID)
 }
+
+// Status queries the status of the client by clientID
+func (s *UpgradeTestSuite) Status(ctx context.Context, chain ibc.Chain, clientID string) (string, error) {
+	queryClient := s.GetChainGRCPClients(chain).ClientQueryClient
+	res, err := queryClient.ClientStatus(ctx, &clienttypes.QueryClientStatusRequest{
+		ClientId: clientID,
+	})
+	if err != nil {
+		return "", err
+	}
+
+	return res.Status, nil
+}
+
+// ClientState queries the current ClientState by clientID
+func (s *UpgradeTestSuite) ClientState(ctx context.Context, chain ibc.Chain, clientID string) (*clienttypes.QueryClientStateResponse, error) {
```

Comment:
fetching client state might be something we want to do in other tests, I'm happy to leave it here for now, but it might be worth moving this to the `E2ETestSuite` WDYT?

Reply:
i could move this one and Status as well, I think they are generic enough functions